src/components/organisms/tag-skill-subsection.astro

```diff
@@ -0,0 +1,36 @@
+---
+import Tag from '@/atoms/tag.astro';
+import type { Tag as TagData } from '@/types/common';
+import type { IconName } from '@/types/icon';
+
+export interface Props {
+  skills: TagData[];
+}
+
+const { skills } = Astro.props;
+
+const getIconProps = (name?: IconName, color?: string) => {
+  if (name) {
+    return {
+      name,
+      color,
+    };
+  }
+
+  return undefined;
+};
+---
+
+<div class:list={['flex', 'flex-wrap', 'gap-3', 'items-center']}>
+  {
+    skills.map(({ name, icon, iconColor, url }) => {
+      return url ? (
+        <a href={url} target="_blank" rel="noopener noreferrer">
+          <Tag icon={getIconProps(icon, iconColor)}>{name}</Tag>
```

Comment:
Może by wewnątrz tego komponentu `Tag` zmienić warunek, żeby nie renderował ikony kiedy nie ma ona nazwy a nie kiedy cały obiekt jest niezdefiniowany. Powinno wyjść na to samo a będzie można pozbyć się tej funkcji `getIconProps`

Reply:
W takim wypadku uznałem, że lepiej będzie wypłaszczyć tą strukturę z:
```ts
export interface Props {
  icon?: {
    name?: IconName;  
    color?: string;
  }
}
```
na: 
```ts
export interface Props {
  name?: IconName;
  color?: string;
}
```